src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenStringConcat.cs

```diff
@@ -650,74 +720,161 @@ private static void TestMethod<T>()
             comp.VerifyDiagnostics();
             comp.VerifyIL("Test.TestMethod<T>()", @"
 {
-  // Code size      211 (0xd3)
-  .maxstack  4
-  .locals init (T V_0)
+  // Code size      453 (0x1c5)
```

Comment:
Still looking at this but it seems weird that the code size increased so much in this test.

Reply:
The null conditional operator on unconstrained generics is pretty wordy ([SharpLab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDZICYgNQA+AAgAwAERAjANwCwAUEQMwUBMZAwmQN4Nn8UWVcgFkAPABUAfAAoJZAGYB7JQEoefAVqIB2MtjgKIAVwww5qgPwA6CUoDKMBGgB2Acxmq69LQF8GvkA==)) and that happens a lot in this method. So each line has gone up from 7-ish instructions to 15-ish, which matches the increase in code size, more or less. 

String concatenation with unconstrained generics is something that this PR makes significantly larger. The JIT should eliminate half of if (according to [this comment](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/CodeGen/EmitExpression.cs,404)), but that doesn't help the assembly size.